util/conditions/patch.go

```diff
@@ -116,10 +124,14 @@ func WithForceOverwrite(v bool) ApplyOption {
 // Apply executes a three-way merge of a list of Patch.
 // When merge conflicts are detected (latest deviated from before in an incompatible way), an error is returned.
 func (p Patch) Apply(latest Setter, options ...ApplyOption) error {
-	if len(p) == 0 {
+	if p.IsZero() {
 		return nil
 	}
 
+	if util.IsNil(latest) {
+		return errors.New("error patching conditions: object was nil")
```

Comment:
```suggestion
		return errors.New("error patching conditions: latest was nil")
```

Reply:
I think object might work better here, just because it's not very clear what 'latest' means outside of the context of the code once this error gets returned to the user. If you'd prefer I can go this way though.